### PR TITLE
fix: QueryBuilder breaks select when escape is false

### LINKED ISSUE #4311: Bug: [QueryBuilder] escaping select string
escaping a query select is not setting the proper quotes:
```php
$builder = $this->db->table('test_table')
                ->select('"numericValue1" + "numericValue2" AS "numericResult"', false)
                ->get();
echo (string)$this->db->getLastQuery();
```

that previous code return this:
```sql
SELECT numericValue1" + "numericValue2 AS "numericResult"
FROM "test_table"
```

this also happens with subselects, CASE WHEN statement and every function with ','.

### LINKED ISSUE #4355: Bug: [QueryBuilder] unable to use regular expression inside a query using module class
**Describe the bug**
If you use a regular expression inside a query C4 add ` and space unwanted.

**CodeIgniter 4 version**
^4

**Affected module(s)**
CodeIgniter\Model

**Expected behavior, and steps to reproduce if appropriate**
If you write a query like this:

```PHP
 $result =   $this->select('REGEXP_SUBSTR(ral_anno,"[0-9]{1,2}([,.][0-9]{1,3})([,.][0-9]{1,3})") AS ral')
                    ->where("period", trim($period)."-01-01")
                    ->where("livello", trim($livello))
                    ->orderBy("ral", "DESC")
                    ->limit(1)
                    ->find();
```

the final result is:

```SQL
SELECT REGEXP_SUBSTR(ral_anno, "[0-9]{1, 2}([, .`][0-9]{1`, 3})([, .`][0-9]{1`, 3})") AS ral
FROM `ob_human_resources`
WHERE `period` = '2018-01-01'
AND `livello` = 'V'
ORDER BY `ral` DESC
```

The problem is in regular expression:

[0-9]{1, 2}([, .`][0-9]{1`, 3})([, .`][0-9]{1`, 3}) -> this is not original one i write.

` and space are added

Following documentation i also try to add second parameter to select

```PHP 
 $result =   $this->select('REGEXP_SUBSTR(ral_anno,"[0-9]{1,2}([,.][0-9]{1,3})([,.][0-9]{1,3})") AS ral', false)
                    ->where("period", trim($period)."-01-01")
                    ->where("livello", trim($livello))
                    ->orderBy("ral", "DESC")
                    ->limit(1)
                    ->find();
```
the result is no ` and it's ok but space are also added:

```SQL
SELECT REGEXP_SUBSTR(ral_anno, "[0-9]{1, 2}([, .][0-9]{1, 3})([, .][0-9]{1, 3})") AS ral
FROM `ob_human_resources`
WHERE `period` = '2018-01-01'
AND `livello` = 'V'
ORDER BY `ral` DESC
```

**Context**
 - OS: Windows 10
 - Web server Apache2
 - PHP version 7.3

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
fixes #4311
fixes #4355

I'm not sure correct behavior of `protectIdentifiers()`. See [tests](https://github.com/codeigniter4/CodeIgniter4/pull/5118/files#diff-65d39fd097198820866d41c227f1afbb8db3498670ad049d13f87ad6df2cc97e).

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Some tests on SQLSRV fail. Why?

```
There were 4 errors:

1) CodeIgniter\Models\MiscellaneousModelTest::testReplaceObject
CodeIgniter\Database\Exceptions\DatabaseException: Failed to get index data from database.

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:221
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:1430
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:371
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:347
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Model.php:397
/home/runner/work/CodeIgniter4/CodeIgniter4/system/BaseModel.php:1038
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Models/MiscellaneousModelTest.php:80

2) CodeIgniter\Models\ReplaceModelTest::testReplaceRespectsUseTimestamps
CodeIgniter\Database\Exceptions\DatabaseException: Failed to get index data from database.

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:221
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:1430
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:371
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:347
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Model.php:397
/home/runner/work/CodeIgniter4/CodeIgniter4/system/BaseModel.php:1038
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Models/ReplaceModelTest.php:36

3) CodeIgniter\Database\Live\InsertTest::testReplaceWithNoMatchingData
Exception: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]The object 'dbo.db_"db_job"' does not exist in database 'test' or is invalid for this operation.

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:459
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:663
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:591
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:220
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:1430
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:371
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:347
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Database/Live/InsertTest.php:69

4) CodeIgniter\Database\Live\InsertTest::testReplaceWithMatchingData
Exception: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]The object 'dbo.db_"db_job"' does not exist in database 'test' or is invalid for this operation.

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:459
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:663
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:591
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:220
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:1430
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:371
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Builder.php:347
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Database/Live/InsertTest.php:86
```

File: system/Database/BaseConnection.php [hunk 1]
```diff
@@ -994,6 +995,11 @@ public function protectIdentifiers($item, bool $prefixSingle = false, ?bool $pro
             return $item;
         }
 
+        // Do not protect identifiers and do not prefix, no swap prefix, there is nothing to do
+        if ($protectIdentifiers === false && $prefixSingle === false && $this->swapPre === '') {
```
Comment (iRedds) [inline system/Database/BaseConnection.php:999]: If I'm not mistaken, if the swapPre property is set, escaping is unavoidable.

Comment (kenjis) [reply system/Database/BaseConnection.php:999]: What do you mean? Is there something wrong with the `if` conditions?

Comment (kenjis) [reply system/Database/BaseConnection.php:1006]: If the swapPre property is set (not empty string), the `if` condition will be false.

**Comment by kenjis** [comment]
I hope someone could review this.

**Comment by MGatner** [comment]
@michalsn or @paulbalandan or @samsonasik? I will if I nobody else does.

**Review by michalsn — APPROVED** [review]

**Review by MGatner — COMMENTED** [review]
I don't know how to answer the questions about the fringe test cases. I assume what you have there is "how it was already working", or some logical outcome of this fix. Maybe a Slack or forum discussion would be helpful? I am not good with the nuances of the database layer. 

And: thanks @michalsn for the review!

**Comment by kenjis** [comment]
@michalsn Thank you for the review!

@MGatner Thank you for the comment.

> Maybe a Slack or forum discussion would be helpful?

I'm not sure, but I posted.

**Comment by kenjis** [comment]
This PR was too old, so I rebased and updated some comments.

**Pull request merged**